File: php/features.php
<?php
declare(strict_types=1);

// SetupGearGuide SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SetupGearGuideFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SetupGearGuideBaseFeature();
            case "test":
                return new SetupGearGuideTestFeature();
            default:
                return new SetupGearGuideBaseFeature();
        }
    }
}

File: py/features.py
# SetupGearGuide SDK feature factory

from feature.base_feature import SetupGearGuideBaseFeature
from feature.test_feature import SetupGearGuideTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SetupGearGuideBaseFeature(),
        "test": lambda: SetupGearGuideTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
